ui/apps/platform/src/Containers/DeferralConfiguration/useVulnerabilitiesDeferralConfig.ts

```diff
@@ -0,0 +1,36 @@
+import { useCallback } from 'react';
+
+import useRestMutation, { UseRestMutationReturn } from 'hooks/useRestMutation';
+import useRestQuery from 'hooks/useRestQuery';
+import {
+    VulnerabilitiesDeferralConfig,
+    fetchVulnerabilitiesDeferralConfig,
+    updateVulnerabilitiesDeferralConfig,
+} from 'services/DeferralConfigService';
+
+export type UseVulnerabilitiesDeferralConfigReturn = {
+    config: VulnerabilitiesDeferralConfig | undefined;
+    isConfigLoading: boolean;
+    configLoadError: unknown;
+    isUpdateInProgress: boolean;
+    updateConfig: UseRestMutationReturn<
+        Partial<VulnerabilitiesDeferralConfig>,
+        Partial<VulnerabilitiesDeferralConfig>
+    >['mutate'];
+};
+
+export function useVulnerabilitiesDeferralConfig(): UseVulnerabilitiesDeferralConfigReturn {
+    const fetchConfigFn = useCallback(fetchVulnerabilitiesDeferralConfig, []);
```

Comment:
@dvail Is it necessary to do this rather than passing `useVulnerabilitiesDeferralConfig` as an argument to the `useRestQuery`?

Reply:
Ah good catch, I've been using `useCallback` with this hook out of muscle memory but in this case is isn't necessary.